app/assets/javascripts/admin/util.js.erb

```diff
@@ -1,6 +1,6 @@
 $(document).ready(function() {
   $('.datetimepicker').datetimepicker({
-    dateFormat: Spree.translations.date_picker,
+    dateFormat: "yy-mm-dd",
```

Comment:
That's weird that the translation doesn't fall back to the one defined in `en.yml`...